README.md

```diff
@@ -3,163 +3,277 @@
+* [CUDA Toolkit >=6.0](https://developer.nvidia.com/cuda-toolkit-60) [Optional, for GPU support]
 
+In addition to this, you will need other [NEURON dependencies](https://github.com/neuronsimulator/nrn) like Python, Flex, Bison etc.
 
 ## Installation
 
-This project uses git submodules which must be cloned along with the repository itself:
+CoreNEURON is now integrated into development version of NEURON simulator. If you are a NEURON user, preferred way to install CoreNEURON is to enable extra build options during NEURON installation as follows:
 
-```
-git clone --recursive https://github.com/BlueBrain/CoreNeuron.git
+1. Clone latest version of NEURON:
 
-```
+  ```
+  git clone https://github.com/neuronsimulator/nrn
+  cd nrn
+  ```
 
-Set the appropriate MPI wrappers for the C and C++ compilers, e.g.:
+2. Create a build directory:
 
-```bash
-export CC=mpicc
-export CXX=mpicxx
-```
+  ```
+  mkdir build
+  cd build
+  ```
 
-And build using:
+3. Load modules : Currently CoreNEURON version rely on compiler auto-vectorisation and hence we advise to use Intel/Cray/PGI compiler to get better performance. This constraint will be removed in near future with the integration of [NMODL](https://github.com/BlueBrain/nmodl) project. Load necessary modules on your system, e.g.
+
+	```
+	module load intel intel-mpi python cmake
+	```
+If you are building on Cray system with GNU toolchain, set following environmental variable:
+
+	```bash
+	export CRAYPE_LINK_TYPE=dynamic
+	```
+
+3. Run CMake with the appropriate [options](https://github.com/neuronsimulator/nrn#build-using-cmake) and additionally enable CoreNEURON with `-DNRN_ENABLE_CORENEURON=ON` option:
+
+  ```bash
+  cmake .. \
+   -DNRN_ENABLE_CORENEURON=ON \
+   -DNRN_ENABLE_INTERVIEWS=OFF \
+   -DNRN_ENABLE_RX3D=OFF \
+   -DCMAKE_INSTALL_PREFIX=$HOME/install
+  ```
+If you would like to enable GPU support with OpenACC, make sure to use `-DCORENRN_ENABLE_GPU=ON` option and use PGI compiler with CUDA.
+> NOTE : if you see error and re-run CMake command then make sure to remove temporary CMake cache files by deleting `CMakeCache.txt`.
+
+4. Build and Install :  once configure step is done, you can build and install project as:
+
+	```bash
+	make -j
+	make install
+	```
+
+## Building Model
+
+Once NEURON is installed with CoreNEURON support, setup `PATH` and `PYTHONPATH ` variables as:
 
-```bash
-cmake ..
-make -j
+```
+export PYTHONPATH=$HOME/install/lib/python:$PYTHONPATH
+export PATH=$HOME/install/bin:$PATH
 ```
 
-If you don't have MPI, you can disable MPI dependency using the CMake option `-DCORENRN_ENABLE_MPI=OFF`:
+Like typical NEURON workflow, you can use `nrnivmodl` to translate MOD files:
 
-```bash
-export CC=gcc
-export CXX=g++
-cmake .. -DCORENRN_ENABLE_MPI=OFF
-make -j
+```
+nrnivmodl mod_directory
 ```
 
-And you can run inbuilt tests using:
+In order to enable CoreNEURON support, we have to use `-coreneuron` flag:
```

Comment:
This has now become confusing as it is a little unclear whether ```nrnivmodl -coreneuron ...``` also builds the special for NEURON.  I thought ``` nrnivmodl ...``` would do both.

Reply:
if coreneuron support is enabled then:
* nrnivmodl : same behaviour as it is i.e. build only special
* nrnivmodl -coreneuron : calls nrnivmodl-core to build special-core followed by nrnivmodl to build special

I kept this behaviour so that user can choose whether to enable coreneuron (because mod files may not compile with nrnivmodl-core)